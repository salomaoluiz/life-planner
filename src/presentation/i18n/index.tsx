import React from "react";

import {
  reactI18NHooks,
  ReactI18NPProvider,
  reactI18NTranslate,
} from "@presentation/i18n/react-i18n";

interface Props {
  children: React.ReactNode;
}

export function I18NProvider({ children }: Props) {
  return <ReactI18NPProvider>{children}</ReactI18NPProvider>;
}

export function translate(key: string, params?: Record<string, string>) {
  return reactI18NTranslate(key, params);
}

export function useTranslation() {
  return reactI18NHooks.useTranslation();
}

export function useTranslationLocale() {
  return reactI18NHooks.useTranslationLocale();
}
