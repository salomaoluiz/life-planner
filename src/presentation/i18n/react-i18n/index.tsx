export * as reactI18NHooks from "./hooks";
export { ReactI18NPProvider } from "./provider";
