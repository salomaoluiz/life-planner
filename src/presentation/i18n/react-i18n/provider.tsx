import React from "react";
import i18next from "i18next";

import { I18nextProvider } from "react-i18next";
import { useI18NInitializer } from "./hooks";

interface Props {
  children: React.ReactNode;
}

export function ReactI18NPProvider({ children }: Props) {
  useI18NInitializer();
  return <I18nextProvider i18n={i18next}>{children}</I18nextProvider>;
}
