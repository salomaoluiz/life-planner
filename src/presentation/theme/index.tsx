export { useTheme } from "./hook";
export { ThemeProvider } from "./provider";
