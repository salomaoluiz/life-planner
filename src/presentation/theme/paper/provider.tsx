import React from "react";
import { PaperProvider } from "react-native-paper";
import { ThemeProp } from "react-native-paper/lib/typescript/types";

interface Props {
  children: React.ReactNode;
  theme: ThemeProp;
}

const PaperThemeProvider = ({ children, theme }: Props) => {
  return <PaperProvider theme={theme}>{children}</PaperProvider>;
};

export default PaperThemeProvider;
