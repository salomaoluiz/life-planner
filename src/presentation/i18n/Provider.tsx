import React from "react";

import { ReactI18NPProvider } from "./react-i18n";

interface Props {
  children: React.ReactNode;
}

function I18NProvider({ children }: Props) {
  return <ReactI18NPProvider>{children}</ReactI18NPProvider>;
}

export default I18NProvider;
