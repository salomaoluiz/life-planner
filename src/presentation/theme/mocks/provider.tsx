import { View } from "react-native";
import * as reactNative from "react-native";

import { ThemeProvider } from "@presentation/theme";
import * as paper from "@presentation/theme/paper";
import * as loader from "@providers/loader";
import { render } from "@tests";

jest.mock("@presentation/theme/paper");
jest.unmock("@presentation/theme");

const providerLoaderResult = {
  isLoading: false,
  setIsLoading: jest.fn(),
};

jest
  .spyOn(paper, "PaperThemeProvider")
  .mockImplementation(({ children, ...props }) => (
    <View testID={"paper-theme-provider"} {...props}>
      {children}
    </View>
  ));
jest.spyOn(reactNative, "useColorScheme").mockReturnValue("light");

jest.spyOn(loader, "useProviderLoader").mockReturnValue(providerLoaderResult);
const defaultProps = {
  children: <View testID={"default-children"} />,
};

function renderComponent() {
  return (
    <ThemeProvider {...defaultProps}>{defaultProps.children}</ThemeProvider>
  );
}

function setup() {
  render(renderComponent());
}

export { providerLoaderResult, setup };
