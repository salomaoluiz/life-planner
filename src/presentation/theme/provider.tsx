import React, { createContext, useEffect, useState } from "react";
import { useColorScheme } from "react-native";

import { useProviderLoader } from "@providers/loader";

import { colors, getScaledSizes } from "./constants";
import { PaperThemeProvider } from "./paper";
import { ThemeProp } from "./types";

interface Props {
  children: React.ReactNode;
}

interface ThemeContextData {
  isDark: boolean;
  setIsDark: (isDark: boolean) => void;
}

export const lightTheme: ThemeProp = {
  colors: colors.light,
  dark: false,
  sizes: getScaledSizes(),
};

const darkTheme: ThemeProp = {
  colors: colors.dark,
  dark: true,
  sizes: getScaledSizes(),
};

export const ThemeContext = createContext<ThemeContextData>(
  {} as ThemeContextData,
);

export function ThemeProvider({ children }: Props) {
  const { setIsLoading } = useProviderLoader();
  const colorSchema = useColorScheme();

  const [isDark, setIsDark] = useState(colorSchema === "dark");
  const [theme, setTheme] = useState<ThemeProp>(
    isDark ? darkTheme : lightTheme,
  );

  useEffect(() => {
    setTheme(isDark ? darkTheme : lightTheme);
    setIsLoading(false, "theme");
  }, [isDark]);

  return (
    <ThemeContext.Provider value={{ isDark, setIsDark }}>
      <PaperThemeProvider theme={theme}>{children}</PaperThemeProvider>
    </ThemeContext.Provider>
  );
}
