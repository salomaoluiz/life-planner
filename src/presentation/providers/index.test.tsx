import * as i18n from "@presentation/i18n";
import * as theme from "@presentation/theme/provider";
import { View } from "react-native";
import { render, screen } from "@tests";
import PresentationProviders from "./";

jest
  .spyOn(i18n, "I18NProvider")
  .mockImplementation(({ children }) => (
    <View testID={"i18n-provider"}>{children}</View>
  ));
jest
  .spyOn(theme, "ThemeProvider")
  .mockImplementation(({ children }) => (
    <View testID={"theme-provider"}>{children}</View>
  ));

function Children() {
  return <View testID={"presentation-provider-children"}></View>;
}

function setup() {
  render(
    <PresentationProviders>
      <Children />
    </PresentationProviders>,
  );
}

it("SHOULD render the Providers", () => {
  setup();

  expect(screen.getByTestId("i18n-provider")).toBeTruthy();
  expect(screen.getByTestId("theme-provider")).toBeTruthy();
  expect(screen.getByTestId("presentation-provider-children")).toBeTruthy();
  expect(screen.toJSON()).toMatchSnapshot();
});
