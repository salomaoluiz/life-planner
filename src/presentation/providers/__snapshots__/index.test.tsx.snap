// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SHOULD render the Providers 1`] = `
<View
  testID="theme-provider"
>
  <View
    testID="i18n-provider"
  >
    <View
      testID="presentation-provider-children"
    />
  </View>
</View>
`;
